wordpress-mobile/WordPress-Android
# [Nullability Annotations to Java Classes] Replace `findViewById` with `ViewBinding` for Reader Comments (`relatively safe`)

Parent #18906
Similar To #19217

This PR's main focus is to replace `findViewById ` with `ViewBinding` for the [ReaderCommentListActivity.java](https://github.com/wordpress-mobile/WordPress-Android/blob/a4fa6ac15e45c3b616dc86d09529584fdd08c971/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderCommentListActivity.java#L4) class.

Additionally, this PR is also dealing with adding any missing nullability annotations to this activity class, focusing mainly on its fields, and, any effected neighbour classes. But first, all existing warnings on this class are being resolved/suppressed so as to make it easier to deal with any missing nullability checks (with help from the IDE), especially when a `@Nullable` annotation is added.

FYI: This change is `relatively safe`, meaning that although there are compile-time changes associated with this change, and it needs testing, the changes included in this PR are very targeted and specific to one screen, and one screen only. Thus, if this screen is tested appropriately, it should be safe to merge this to `trunk`.

-----

PS: @AjeshRPai I added you as the main reviewer, not so randomly, due to the fact that you've also reviewed #19217, and I wanted you from the Jetpack/WordPress mobile team again to be aware of and sign-off on that change for JP/WPAndroid. Feel free to merge this PR directly yourself if you deem so.

-----

`findViewById ` -> `ViewBinding`

1. [Replace findviewbyid with viewbinding for reader comment list](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/351335e39e82d5d31dabf2483220f9a32fb172b9)
2. [Use non-null vb method param for reader comment list (1/2)](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/5811448399dceb1e28c413ca2c4c26861fa4aac0)
3. [Use non-null vb method param for reader comment list (2/2)](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/d9fa586723f99e2622166941eaa27352190f443c)

Nullability Annotation List:

1. [Add missing nullability annotations to sdk override methods](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/2c3f0ab44b4fb7215eae36b4d1a2e9bb29def708)
2. [Add missing nl-a to post field](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/896dfc58bb697ce8efc724e2a2d788073b48a1bb)
3. [Add missing nl-a to comment adapter field](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/65e6c1c2d598463c7bc58eeda21456a72ae73a26)
4. [Add missing nl-a to suggestion adapter field](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/a50d2e1e286280444c631ff6b361f60ca3b250d2)
5. [Add missing nl-a to suggestion service connection mgr field](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/ca1a478878fa19ccda72ba5bfa25b0aea573042a)
6. [Add missing nl-a to swipe to refresh helper field](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/96894dbe0186da437c5bbec78fbf7125f5c988d9)
7. [Add missing nl-a to direct operation field](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/3385b1124b29ceedfb0021fc8691dcea31ac1759)
8. [Add missing nl-a to intercepted uri field](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/31636b22a9545763646d328362382d8c40896854)
9. [Add missing nl-a to source field](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/823603288dd148cecf9f21d22cab05b8340e82af)
10. [Add missing nl-a to view model field](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/b9a8dcfb1ef450fb78a2c3db81f21627893145d9)
11. [Add missing nl-a to conversation view model field](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/8fecb2bd1f79deca7851af609408067094404b38)
12. [Add missing nl-a to saved instance state parameter](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/ce75e86e902723d6bf0485f97d3d0914fc147678)

Warnings Resolution List:

1. [Replace unnecessary switch with if stmt on reader cmnt list](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/3aacd9aa1a8501278fd0d38d175f52b77f4cc0e8)
2. [Guard nullable post site variable on reader comment list](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/b85afb881e57d82dbc47a9dde84c646305e71eaf)
3. [Guard nullable method invocation on reader comment list](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/0d7034ee5a38c06aa3e86ec7437e564a8a3c4309)
4. [Merge divider no action with catch approve on reader cmnt list](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/9c199c5714b69613cd1eab03a632483de7df220a)

Warnings Suppression List:

1. [Suppress notify data set changed lint warn on reader cmnt list](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/311509ee3ed18525e1317cc9dd399100fe541d10)
2. [Suppress same parameter value warning on reader comment list](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/017907cbb68c0fb036b92aeb0e4262e6a501f403)
3. [Suppress deprecation warning on reader comment list](https://github.com/wordpress-mobile/WordPress-Android/pull/19333/commits/f5558913e99ccb3a9f664efe9d5f6777c97a958f)

-----

## To test:

1. Quickly smoke test any reader comments related functionality on the Jetpack app and see if everything is working as expected.
2. In addition to the above smoke test, you can expand the below and follow the inner and more explicit test steps within:

<details>
    <summary>Reader Comments Screen [ReaderCommentListActivity.java]</summary>

ℹ️ This test applies to the `Jetpack` app only.

- Go to `Reader` tab, then its `DISCOVER` sub-tab, and tap on any post.
- Tap on the `Comments` button at the bottom of the post.
- Verify that the `Reader Comments` screen is shown and functioning as expected.
- For example, try (un)following, enabling/disabling in-app notifications, replying to post, and then sharing, replying or liking a specific comment.

</details>

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

16. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

19. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR submission checklist:


-----

## UI Changes testing checklist:
